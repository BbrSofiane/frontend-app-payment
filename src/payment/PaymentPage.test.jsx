/* eslint-disable global-require */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { IntlProvider, configure as configureI18n } from '@edx/frontend-i18n';
import configureMockStore from 'redux-mock-store';

import * as analytics from '@edx/frontend-analytics';
import ConnectedPaymentPage from './PaymentPage';
import { configuration } from '../environment';
import messages from '../i18n';

const mockStore = configureMockStore();
const storeMocks = {
  loadingApp: require('./__mocks__/loadingApp.mockStore.js'),
};
const requirePaymentPageProps = {
  fetchBasket: () => {},
};

// Mock language cookie
Object.defineProperty(global.document, 'cookie', {
  writable: true,
  value: `${configuration.LANGUAGE_PREFERENCE_COOKIE_NAME}=en`,
});

configureI18n(configuration, messages);

describe('<PaymentPage />', () => {
  describe('Renders correctly in various states', () => {
    it('app loading', () => {
      analytics.sendTrackingLogEvent = jest.fn();
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <Provider store={mockStore(storeMocks.loadingApp)}>
              <ConnectedPaymentPage {...requirePaymentPageProps} />
            </Provider>
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
