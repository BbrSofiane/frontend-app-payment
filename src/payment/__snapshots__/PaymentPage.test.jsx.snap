// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentPage /> Renders correctly in various states should render errors 1`] = `
<div
  className="page__payment container-fluid py-5"
>
  <h1>
    Payment
  </h1>
  <div>
    Error: Something went really wrong.
  </div>
</div>
`;

exports[`<PaymentPage /> Renders correctly in various states should render its default state 1`] = `
<div
  className="page__payment container-fluid py-5"
>
  <h1>
    Payment
  </h1>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <p>
        Your basket is empty.
      </p>
    </div>
    <div
      className="col-6"
    />
  </div>
</div>
`;

exports[`<PaymentPage /> Renders correctly in various states should render its loading state 1`] = `
<div
  className="page__payment container-fluid py-5"
>
  <h1>
    Payment
  </h1>
  <div>
    <div
      className="d-flex justify-content-center align-items-center flex-column"
      style={
        Object {
          "height": "50vh",
        }
      }
    >
      <div
        className="spinner-border text-primary"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading basket...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PaymentPage /> Renders correctly in various states should render the basket 1`] = `
<div
  className="page__payment container-fluid py-5"
>
  <h1>
    Payment
  </h1>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <ul>
        <li>
          showVoucherForm: 
          false
        </li>
        <li>
          paymentProviders: 
        </li>
        <li>
          orderTotal: 
          10
        </li>
        <li>
          lineDiscount: 
        </li>
        <li>
          sdnCheck: 
          false
        </li>
        <li>
          lineTotal: 
        </li>
        <li>
          products: 
        </li>
        <li>
          voucher: 
        </li>
      </ul>
    </div>
    <div
      className="col-6"
    />
  </div>
</div>
`;
